src/store/getters/state-getters.ts

```diff
@@ -1029,3 +1060,13 @@ export const getSpecialResolutionConfirmValid = (state: StateIF): boolean => {
 export const getRestoration = (state: StateIF): RestorationStateIF => {
   return state.stateModel.restoration
 }
+/** Returns true when users can change the sole proprietor (SP).
+ * Restricts the ability of non-staff users from changing the sole
+ * proprietor when the SP is a non-individual.  This restriction has been
+ * added to ensure that changes made in business-edit-ui are also updated by
+ * staff in COLIN */
+export const showSoleProprietorChangeButton = (state: StateIF, getters): boolean => {
+  const isProprietor = getters.getOrgPeople?.[0].roles?.[0]?.roleType === RoleTypes.PROPRIETOR
+  const isOrganization = getters.getOrgPeople?.[0]?.officer?.partyType === PartyTypes.ORGANIZATION
+  return !(!getters.isRoleStaff && getters.isSoleProp && isOrganization && isProprietor)
```

Comment:
Please simplify this negative-negative logic. It may be better to use a variable for the second part (isSoleProp && isOrganization && isProprietor) and then return "isRoleStaff && isNewVariable".

Another simplification: if this is a SP then the officer must necessarily be a proprietor. (That's the definition of Proprietor. Likewise, a Partner only exists for a GP.)

Reply:
Good points, I'll simplify it as you propose.

Reply:
I really like the simplification you proposed!